lib/ops/opsservice/agents.go

```diff
@@ -581,6 +566,47 @@ func (r *AgentPeerStore) addGroup(key ops.SiteOperationKey) (*agentGroup, error)
 	return agentGroup, nil
 }
 
+func (r *AgentPeerStore) isPartOfActiveOperation(addr string, token storage.ProvisioningToken) error {
+	op, err := r.backend.GetSiteOperation(token.SiteDomain, token.OperationID)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	operation := (ops.SiteOperation)(*op)
+	if operation.IsCompleted() {
+		return trace.BadParameter("operation %v is already completed", token.OperationID)
+	}
+	serverAddr := utils.ExtractHost(addr)
+	if (storage.Servers)(op.Servers).FindByIP(serverAddr) == nil {
```

Comment:
This is only used for install/expand, right? Because IIRC not all operations populate Servers field.